File: 프로그래머스/더맵게.java
import java.util.*;
public class 더맵게 {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq=new PriorityQueue<>();
        for (int s:scoville){
            pq.add(s);
        }
        if (pq.peek()>=K){
            return 0;
        }
        while (pq.size()>=2 && !check(pq,K)){
            int a=pq.poll();
            int b=pq.poll();
            int newScoville=a+2*b;
            pq.add(newScoville);
            answer++;
        }
        
        if(check(pq,K)){
            return answer;
        }
        else{
            return -1;    
        }
        
    }
    boolean check(PriorityQueue<Integer> pq,int K){
        for (int n:pq){
            if (n<K){
                return false;
            }
        }
        return true;
    }
}

File: 프로그래머스/더맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville);
    if scoville[0]>=K:
        return 0;
    
    def check():
        for s in scoville:
            if s<K:
                return False
        return True
    while len(scoville)>=2 and not check():
        a=heapq.heappop(scoville)
        b=heapq.heappop(scoville)
        heapq.heappush(scoville,a+2*b)
        answer+=1
    if check():
        return answer
    else: return -1